src/Engine/ProtoCore/RuntimeData.cs

```diff
@@ -344,10 +357,32 @@ private string GetAndRemoveTraceDataForNode(Guid nodeGuid, string callsiteID)
             {
                 return GetAndRemoveTraceDataForNode(nodeGuid, string.Empty);
             }
-            else
+
+            return callsiteTraceData;
+        }
+
+        private static bool DoCallSiteIDsMatch(string compilerGenerated, string deserialized)
+        {
+            if (compilerGenerated == deserialized) return true;
+
+            var matches1 = Regex.Match(compilerGenerated, fullCallsiteID);
+            var matches2 = Regex.Match(deserialized, fullCallsiteID);
+
+            if (matches1.Groups.Count != matches2.Groups.Count) return false;
+
+            // If both group counts match, they should number 12 in all.
+            // We should ignore checking for the 1st, 7th, and 10th group specifically
+            // as per the Regex pattern (for fullCallsiteID) since that group includes the function scope
```

Comment:
can you provide a sample here -
I'm having a hard time understanding what the 1st,7th,or 9th entry refer to?

Reply:
I guess my reason for wanting to understand this better is to be able to ask more specific questions like - without these 3 groups, can we sufficiently identify these trace data entries from each other and definitely associate them with the correct nested node?

Reply:
@mjkkirschner had some time to answer you today. See the following example for a callsite identifier for a custom node:
![image](https://user-images.githubusercontent.com/5710686/54729780-ccb48300-4b5b-11e9-9a0d-218a03273bc9.png)

In this example, the deserialized and compiler generated ID's match in every respect except for the function scope of the nested node. The function scope generated by an older version of Dynamo, which is serialized in the old DYN file is 110, while the function index of the same custom node when loaded in the latest Dynamo version (compiler generated) is 107. 

If you see the function groups, there are 12 of them broken down as shown. You can see that the 1st, 7th and 10th groups have the function scope number in them (107 in this example), which need to be ignored. This is based on the way that the Regex (funcCallsiteID) is generated.

Reply:
@aparajit-pratap thanks for following up! 😄 
the team will look at finishing this one up ASAP.